EVA-1586 First version of the pipeline (#1)
EBIvariation/variant-remapping: README.md

# variant-remapping

> ## Description

Reviewer: For consistency with other README files.
```suggestion
```

> The pipeline for remapping VCF variants between two arbitrary FASTA assemblies. No chain file required.

Reviewer: ```suggestion
Pipeline for remapping VCF variants between two arbitrary assemblies in FASTA format. No chain file is required.
```

**Method**: creates reads from the flanking sequences of each variant, then maps them to the new assembly using bowtie2.

Currently, it only supports SNPs, and not indels.

Prerequisites:

> - reverse_strand.py, included (reverse strand allele correction, uses bamnostic)

Reviewer: Does this mean the alleles are converted to the forward strand? Could it be written down that way, for the not-so-biologically savvy (most people in the team 😂)?

Author: Yes! I've added an explanation in the README file.

- write_header.py, included (necessary for VCF header generation)
- vcf2bed
- bowtie2
- samtools
- bedtools
- bcftools

## Input

> - Old genome assembly file (fasta format): the genome you have variants for.

Reviewer: ```suggestion
- Old genome assembly file (FASTA format): the genome you have variants for.
```

> - New genome assembly file (fasta format): the genome you want to remap the variants to.

Reviewer: ```suggestion
- New genome assembly file (FASTA format): the genome you want to remap the variants to.
```

> - Variants file (vcf format): contains the list of variants you want to remap.

Reviewer: ```suggestion
- Variants file (VCF format): contains the list of variants you want to remap.
```

Command usage: (see Input for details)
`./remapping_commands.sh -g [oldgenome] -n [newgenome] -a [newgenomeaccession] -v [vcffile] -o [outfile name]`

You can use `time` at the beginning of the previous command to get the runtime and CPU time.

**Example command:**
`time ./remapping_commands.sh -g droso_dm3.fasta -n droso_dm6.fasta -a GCA_000001215.4 -v droso_variants_renamed.vcf -o test.vcf`
**Output:**
```
-----------------------1) Flanking sequence generation-----------------------
---------------------------2) Mapping with bowtie2---------------------------
Mapping results:
1077 reads; of these:
  1077 (100.00%) were unpaired; of these:
    7 (0.65%) aligned 0 times
    48 (4.46%) aligned exactly 1 time
    1022 (94.89%) aligned >1 times
99.35% overall alignment rate
------------------------------3) Data extraction-----------------------------
-----------------------------------Results-----------------------------------
Total number of input variants:
1094
Total number of variants processed (after filters):
1077
Number of remapped variants:
1070
Percentage of remapped variants:
99.300%

real	0m1.363s
user	0m1.091s
sys	0m0.171s
```

> ## Task list

Reviewer: Shall we create tickets in the backlog? In terms of tracking and planning, it is preferable over a list in a plain text file :)

Author: Yes that would be best :) Should I make them?

Reviewer 2: Since at this point we don't have a definitive long-term roadmap for the pipeline, I think it's too soon to start creating the tickets. The only thing that we'll surely have to implement (during @mistyskye's internship or after it) is support for indels. Other points are more planning points than tickets; so I suggest just moving them to the shared Google document so far

Reviewer 2: @mistyskye Could you please remove the task list from the `README` and move it to the Google doc (if not already there)

- Variants file (VCF format): contains the list of variants you want to remap.
- Accession name for the new assembly: required to recreate the header for the output VCF file.

## Output
A VCF file containing:
- remapped coordinates (position on the new assembly)
- rsIDs
- the correct chromosome/contig names
- the new REF alleles (reverse strand mapping taken into account)
- the ALT, QUAL, FILT and INFO columns of the input VCF

Console output: 
```
"Total number of input variants:"
[number]
"Total number of variants processed (after filters):"
[number]
"Number of remapped variants:"
[number]
"Percentage of remapped variants:"
[percentage]
```

## Usage
First, make sure the scripts are executable:
`chmod a+x remapping_commands.sh reverse_strand.py`

Command usage: (see Input for details)
`./remapping_commands.sh -g [oldgenome] -n [newgenome] -a [newgenomeaccession] -v [vcffile] -o [outfile name]`

You can use `time` at the beginning of the previous command to get the runtime and CPU time.

**Example command:**
`time ./remapping_commands.sh -g droso_dm3.fasta -n droso_dm6.fasta -a GCA_000001215.4 -v droso_variants_renamed.vcf \

> 	-o test.vcf`

Reviewer 2: It's better to reformat it to make each flag and its parameter start with the new line, like so:

```bash
time ./remapping_commands.sh \
  -g droso_dm3.fasta \
  -n droso_dm6.fasta \
  -a GCA_000001215.4 \
  -v droso_variants_renamed.vcf \
  -o test.vcf
```

Reviewer 2: (It makes the argument list more readable)

- Variants file (VCF format): contains the list of variants you want to remap.
- Accession name for the new assembly: required to recreate the header for the output VCF file.

## Output
A VCF file containing:
- remapped coordinates (position on the new assembly)
- rsIDs
- the correct chromosome/contig names
- the new REF alleles (reverse strand mapping taken into account)
- the ALT, QUAL, FILT and INFO columns of the input VCF

Console output: 
```
"Total number of input variants:"
[number]
"Total number of variants processed (after filters):"
[number]
"Number of remapped variants:"
[number]
"Percentage of remapped variants:"
[percentage]
```

## Usage
First, make sure the scripts are executable:
`chmod a+x remapping_commands.sh reverse_strand.py`

Command usage: (see Input for details)
`./remapping_commands.sh -g [oldgenome] -n [newgenome] -a [newgenomeaccession] -v [vcffile] -o [outfile name]`

You can use `time` at the beginning of the previous command to get the runtime and CPU time.

**Example command:**
`time ./remapping_commands.sh \
  -g droso_dm3.fasta \
  -n droso_dm6.fasta \
  -a GCA_000001215.4 \
  -v droso_variants_renamed.vcf \

>   -o test.vcf`

Reviewer 2: I don't think this will properly format because single backticks are only meant to be used for inline code formatting. Here it's better to use block code formatting with triple backticks (as shown here https://help.github.com/en/articles/creating-and-highlighting-code-blocks)
